### DISCUSSION TITLE
If all you need is a SPA what are the benefits that next.js brings ?

### DISCUSSION THREAD

**Discussion by geraldodev** [vercel/next.js#39969] [Help]
Hi,

I'm accessing starting with t3-app, and it suggests next.js. 
I have a question: In your opinion, what are the benefits of next js as server side option for a Single Page Application.

Regards,
Geraldo Lopes de Souza

**Comment by icyJoseph** [comment]
- Built-in router
- Prefetching data with Link
- Rendering on the server, also at build time
- Rendering on the client (SPA)
- Fetching data on the server, and sending it to the server to perform client side rendering (SPA)
- Possibility to respond with different content, based on the caller. For example, reading a cookie, or user agent string. 
- API routes to fetch data for your app, from the server. For example, database connections and other's **can be** hidden from the client, and be protected via CORS.
- Lower Time To Visible
- Each route is automatically bundled on its own, no big bundle by default
- Bots, and Non-JS running entities, could still use your app, to some extend
- Control over `html`, and `body` tags, as well as `head` components
- Webpack is under control, for the most part
- TypeScript support

Not sure what your are specifically after though.
